client/src/app/pages/migration-waves/components/manage-applications-form.tsx

```diff
@@ -41,30 +40,47 @@ import { dedupeFunction } from "@app/utils/utils";
 import { useUpdateMigrationWaveMutation } from "@app/queries/migration-waves";
 
 export interface ManageApplicationsFormProps {
+  applications: Application[];
   migrationWave: MigrationWave;
   migrationWaves: MigrationWave[];
   onClose: () => void;
 }
 
 export const ManageApplicationsForm: React.FC<ManageApplicationsFormProps> = ({
+  applications,
   migrationWave,
   migrationWaves,
   onClose,
 }) => {
   const { t } = useTranslation();
   const { pushNotification } = React.useContext(NotificationsContext);
 
-  const { data: applications, isFetching } = useFetchApplications();
-
-  const assignedApplications = migrationWaves
+  const applicationsUsedByOtherMigrationsIds = migrationWaves
     .filter((wave) => wave.id !== migrationWave.id)
     .map((wave) => wave.applications.map((application) => application.id))
     .flat();
 
+  const initialApplicationUsedByMigrationIds = migrationWave.applications.map(
+    (application) => application.id
+  );
+
+  const assignedApplications = applications.filter((application) =>
+    initialApplicationUsedByMigrationIds.includes(application.id)
+  );
+
   const availableApplications = applications.filter(
-    (application) => !assignedApplications.includes(application.id)
+    (application) =>
+      !applicationsUsedByOtherMigrationsIds.includes(application.id)
   );
 
+  const isNewerSelection = (): boolean => {
+    const selectedIds = selectedItems.map((application) => application.id);
+    return (
+      selectedIds.sort().toString() !==
```

Comment:
This is clever! This might be a performance issue in cases where there are hundreds of apps selected, but that doesn't seem likely here right? So this will work fine.

If you want though, here's another solution that will return early as soon as it detects a difference while looping through the arrays. We could export a util for this:
```ts
export const isArrayDifferent = <T>(a: T[], b: T[]) =>
  a.some((val) => !b.includes(val)) || b.some((val) => !a.includes(val));
```

and then the usage here would be
```ts
return isArrayDifferent(selectedIds, initialApplicationUsedByMigrationIds);
```

Whichever you like better is fine, just thinking out loud. I'm actually not sure if this will perform better anyway in the average case. And I'm definitely overthinking it.

Reply:
@mturley, yeah I took the shortcut for a quick implementation but for scalability the array inspection is better.
So let me take that one in.

Reply:
@mturley, I did some tests and the string size blows up before the stack so that's the first limitation of my initial approach. The inspection approach wins anyway being about 140 times faster!